### PR TITLE
Porting: Switch WaitNamedPipe and CreateNamedPipeClient invocation ordering. 

### PR DESCRIPTION
**Pull request by pjanotti**
(Original PR #24616)

* Switch WaitNamedPipe and CreateNamedPipeClient invocation ordering.

We need to do this as WaitNamedPipe is not supported in Windows Containers.
By default using CreateNamedPipeClient first with call to WaitNamedPipe as
a fallback only will allow us to use NamedPipeClientStream inside Windows
Containers.

* Integrate feedback from PR.

### CODE REVIEW

File: src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs [hunk 1]
```diff
@@ -79,14 +56,58 @@ private bool TryConnect(int timeout, CancellationToken cancellationToken)
...
+                    // Server is not yet created
+                    if (errorCode == Interop.Errors.ERROR_FILE_NOT_FOUND)
+                    {
+                        return false;
+                    }
+
+                    // The timeout has expired.
+                    if (errorCode == Interop.Errors.ERROR_SUCCESS)
+                    {
+                        if (cancellationToken.CanBeCanceled)
```
Comment (stephentoub) [inline src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs]: This is the wrong property to check.  It should be `if (cancellationToken.IsCancellationRequested)`, not `if (cancellationToken.CanBeCanceled)`.

Comment (stephentoub) [reply src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs]: I'm also not clear on why the CancellationToken is being consulted at all.  Is code elsewhere Register'ing for cancellation?

File: src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs [hunk 2]
```diff
@@ -79,14 +56,58 @@ private bool TryConnect(int timeout, CancellationToken cancellationToken)
...
+                    errorCode = Marshal.GetLastWin32Error();
+
+                    // Server is not yet created
+                    if (errorCode == Interop.Errors.ERROR_FILE_NOT_FOUND)
+                    {
+                        return false;
+                    }
+
+                    // The timeout has expired.
+                    if (errorCode == Interop.Errors.ERROR_SUCCESS)
```
Comment (stephentoub) [inline src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs]: This PR is changing it from ERROR_SUCCESS to ERROR_SEM_TIMEOUT: 
https://github.com/dotnet/corefx/pull/25197
Should that be done here, too?

**Comment by karelz** [comment]
@pjanotti Is it supposed to fix the 2 bugs linked from the original PR? If yes, please edit PR description to "Fixes #abc", thanks!

**Comment by pjanotti** [comment]
This PR is not going into this servicing branch, I will start a new one and will decide what to do regarding @stephentoub comments

**Pull request closed without merging**